### PR TITLE
feat: New Lw Account Resource Group Terraform Resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Issue: https://lacework.atlassian.net/browse/ALLY-627

New Terraform resource for managing Lacework Account Resource Groups in Lacework

Usage:

``` hcl
resource "lacework_resource_group_account" "lw" {
  name        = "My Lw Account Resource Group"
  description   = "This groups a subset of Lacework accounts"
  accounts = ["my-account"]
}
```

Signed-off-by: Darren Murray darren.murray@lacework.net

### CODE REVIEW

File: integration/integration.go [hunk 1]
```diff
@@ -9,9 +9,11 @@ import (
 )
 
 var LwClient *api.Client
+var LwOrgClient *api.Client
```
Comment (afiune) [inline integration/integration.go:12]: nit
```suggestion
var (
	LwClient    *api.Client
	LwOrgClient *api.Client
)
```

File: examples/resource_lacework_resource_group_lw_account/main.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+provider "lacework" {
+  organization = true
+}
+
+resource "lacework_resource_group_lw_account" "example" {
+  name            = var.resource_group_name
+  description     = var.description
+  accounts        = ["tech-ally"]
```
Comment (afiune) [inline examples/resource_lacework_resource_group_lw_account/main.tf:16]: could we test changing the accounts from this to
```suggestion
  accounts        = ["tech-ally", "mini-ally"]
```
just to verify it applies those changes

File: lacework/provider.go [hunk 3]
```diff
@@ -94,6 +94,7 @@ func Provider() *schema.Provider {
 			"lacework_resource_group_aws":            resourceLaceworkResourceGroupAws(),
 			"lacework_resource_group_azure":          resourceLaceworkResourceGroupAzure(),
 			"lacework_resource_group_gcp":            resourceLaceworkResourceGroupGcp(),
+			"lacework_resource_group_lw_account":     resourceLaceworkResourceGroupLwAccount(),
```
Comment (afiune) [inline lacework/provider.go:97]: I don't think we should add the `lw` into the resource name, could we update it to:
```suggestion
			"lacework_resource_group_account":     resourceLaceworkResourceGroupLwAccount(),
```

File: lacework/resource_lacework_resource_group_lw_account.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of the resource group",
+			},
+			"is_default": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether the resource group is a default resource group.",
```
Comment (afiune) [inline lacework/resource_lacework_resource_group_lw_account.go:77]: nit
``suggestion
				Description: "Whether the resource group is a default resource group",
```

File: website/docs/r/resource_group_lw_account.html.markdown [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Use this resource to create a Lacework Account Resource Group in order to categorize Lacework-identifiable assets.
+For more information, see the [Resource Groups documentation](https://support.lacework.com/hc/en-us/articles/360041727354-Resource-Groups).
+
+## Example Usage
+
+```hcl
+resource "lacework_resource_group_lw_account" "example" {
+  name        = "My Lacework Account Resource Group"
+  description = "This groups a subset of Lacework accounts"
+  accounts = ["my-account"]
```
Comment (afiune) [inline website/docs/r/resource_group_lw_account.html.markdown:20]:
```suggestion
  accounts    = ["my-account"]
```

File: website/docs/r/resource_group_lw_account.html.markdown [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+}
+```
+
+### Organization Level Access
+
+`lacework_resource_group_lw_account` is an organization level integration. And requires `organization` argument to `true` 
+in the provider config.
```
Comment (afiune) [inline website/docs/r/resource_group_lw_account.html.markdown:27]:
```suggestion
The `lacework_resource_group_account` is an organization-level resource that requires the Lacework provider to be configured with the `organization` argument to `true`.

```

**Review by afiune — CHANGES_REQUESTED** [review]

File: lacework/provider.go (hunk 3 above)
Comment (dmurray-lacework) [reply lacework/provider.go:97]: I think the word account is very generic, especially in the context of resource groups.

File: website/docs/r/resource_group_account.html.markdown [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+Use this resource to create a Lacework Account Resource Group in order to categorize Lacework-identifiable assets.
+For more information, see the [Resource Groups documentation](https://support.lacework.com/hc/en-us/articles/360041727354-Resource-Groups).
+
+## Example Usage
+
+```hcl
+resource "lacework_resource_group_account" "example" {
+  name        = "My Lacework Account Resource Group"
+  description = "This groups a subset of Lacework accounts"
+  accounts = ["my-account"]
```
Comment (afiune) [inline website/docs/r/resource_group_account.html.markdown:20]:
```suggestion
  name        = "My Lacework Account Resource Group"
  description = "This groups a subset of Lacework accounts"
  accounts    = ["my-account"]
```

File: website/docs/r/resource_group_account.html.markdown [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+}
+```
+
+### Organization Level Access
+
+`lacework_resource_group_account` is an organization level integration. And requires `organization` argument to `true` 
+in the provider config.
```
Comment (afiune) [inline website/docs/r/resource_group_account.html.markdown:27]:
```suggestion
The `lacework_resource_group_account` is an organization-level resource that requires the Lacework provider to be configured with the `organization` argument to `true`.

```

**Review by afiune — APPROVED** [review]

**Review by afiune — APPROVED** [review]
tenor-41546170

**Comment by afiune** [comment]
Screen Shot 2021-09-24 at 2 40 15 PM

**Pull request merged**